### PR TITLE
Add overflow menu to mini context menu

### PR DESCRIPTION
**Pull request by esh-g**
### Fixed Issues

$ https://github.com/Expensify/App/issues/33573
PROPOSAL: https://github.com/Expensify/App/issues/33573#issuecomment-1875222992

### Tests

1. Send a message in any report
2. Right click and make sure we see the option 'leave thread'
3. Click on 'leave thread'
4. Right click on message again
5. Make sure 'join thread' is displayed.
6. Hover over a report action
7. Make sure it matches the layout below:

Screenshot 2024-01-05 at 8 55 32 PM

8. Make sure the following options are displayed in mini context menu for the specific items.

| Action type | Items |
|--------|--------|
| Task | Reply in thread, mark as unread/read |
| Sent message | Reply in thread, edit action |
| Attachment | Download, reply in thread |
| Received message | Reply in thread, mark as Unread | 
| Money report preview | Reply in Thread, Mark as unread |
| Split bill | Mark as unread, copy to clipboard |
| Money request action | Reply in thread, edit request |

9. Click on the `ThreeDot` menu and make sure the context menu is displayed.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Send a message in any report
2. Right click and make sure we see the option 'leave thread'
3. Click on 'leave thread'
4. Right click on message again
5. Make sure 'join thread' is displayed.
6. Hover over a report action
7. Make sure it matches the layout below:

Screenshot 2024-01-05 at 8 55 32 PM

8. Make sure the following options are displayed in mini context menu for the specific items.

| Action type | Items |
|--------|--------|
| Task | Reply in thread, mark as unread/read |
| Sent message | Reply in thread, edit action |
| Attachment | Download, reply in thread |
| Received message | Reply in thread, mark as Unread | 
| Money report preview | Reply in Thread, Mark as unread |
| Split bill | Mark as unread, copy to clipboard |
| Money request action | Reply in thread, edit request |

9. Click on the `ThreeDot` menu and make sure the context menu is displayed.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/77237602/0cc34e05-2db0-45e9-bcf6-e59ee04f546c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77237602/fadd660d-ce97-41dc-942c-cd2d45ae419b

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77237602/6b73ad6d-a44f-4517-a7fd-1cc1f06d4496

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77237602/bf53c2f2-a5ee-4a2c-8193-03b80f6fe88a

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77237602/ef24f022-9975-4242-9276-82b80cc7fba3

Screenshot 2024-01-11 at 5 17 23 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77237602/1840532d-8be1-4c13-9c35-e81fbc512511

https://github.com/Expensify/App/assets/77237602/0983e61c-602b-4881-8131-8474bc7bd941

</details>

### CODE REVIEW

**Comment by esh-g** [comment]
@rushatgabhane The task of adding the new icons is still pending...
Screenshot 2024-01-05 at 9 22 02 PM
Could you please let me know how we should go about these?

File: src/pages/home/report/ContextMenu/ContextMenuActions.js
Comment (rushatgabhane) [inline src/pages/home/report/ContextMenu/ContextMenuActions.js:488]: Why are we showing / not showing only based on `isMini`?

File: src/pages/home/report/ContextMenu/ContextMenuActions.js
Comment (rushatgabhane) [inline src/pages/home/report/ContextMenu/ContextMenuActions.js:322]: Could you please help me understand why all icons now depend on `isMini`?

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

**Comment by rushatgabhane** [comment]
@esh-g please make sure that the performance tests pass.

**Comment by rushatgabhane** [comment]
> The task of adding the new icons is still pending

we got the icons here! -  https://github.com/Expensify/App/issues/33573#issuecomment-1880323711

File: src/pages/home/report/ContextMenu/ContextMenuActions.js
Comment (esh-g) [reply src/pages/home/report/ContextMenu/ContextMenuActions.js:488]: We only want to show the `ThreeDot` menu in the mini menu and not when we right click...

File: src/components/ContextMenuItem.js
Comment (esh-g) [reply src/components/ContextMenuItem.js:61]: @rushatgabhane To open the context menu when we press the `ThreeDot` menu in the mini context menu, we need to pass the click trigger event. So passing it from here is important.

File: src/pages/home/report/ContextMenu/ContextMenuActions.js
Comment (esh-g) [reply src/pages/home/report/ContextMenu/ContextMenuActions.js:322]: @rushatgabhane For the changes we need to selectively show some option only for `mini` menu. For example, the `ThreeDotMenu` should be only visible when mini context menu is active and not for the full.

**Comment by esh-g** [comment]
@rushatgabhane I have added the icons, fixed the lint and replied to the review comments. Let me know if anything is unclear!

**Comment by rushatgabhane** [comment]
@esh-g could you please add screenshots for all platforms and fix conflicts

File: src/components/Reactions/MiniQuickEmojiReactions.tsx
Comment (rushatgabhane) [inline src/components/Reactions/MiniQuickEmojiReactions.tsx:103]: Please add this value to `style.variable` and use it from there

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
### Bug: For a task, we're showing three dot menu two times.



image


Happens for system messages too. Please fix the root cause of this @esh-g 

image

**Comment by esh-g** [comment]
@rushatgabhane I wanted to ask if the steps in the tests are correct before attaching the test videos... Should I also include tests for things like money reports, attachments, tasks etc?

File: src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js
Comment (rushatgabhane) [inline src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js:83]: the numbers 3 and 4 are very arbitrary. could we please move them to CONST

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
please fix the typescript errors too

**Comment by esh-g** [comment]
@rushatgabhane Could you please confirm if the test steps are right?

**Comment by dubielzyk-expensify** [comment]
Could I get screenshots and videos so I can inspect the pixels please? 😇

**Comment by esh-g** [comment]
@dubielzyk-expensify I have added videos for chrome, here is a screenshot:
Screenshot 2024-01-11 at 5 15 02 PM

Please tell me if any other screenshot/video is needed

**Comment by dubielzyk-expensify** [comment]
This is awesome work and a great start!

Not sure if you're splitting this up into multiple PRs, but I'll give all my feedback here anyways:

- [ ] We're missing new icons for `Reply in thread` and `Mark as unread` on Android and Web. Other platforms look correct
CleanShot 2024-01-12 at 08 53 05@2x

- [ ] `Unsubscribe from thread` should become `Leave thread` on all platforms
- [ ] `Subscribe to thread` should become `Join thread` on all platforms

- [ ] The overflow menu isn't correct on desktop. It shouldn't replicate the overflow menu like mobile, but rather just show the missing options. See mock underneath:
image

**Comment by esh-g** [comment]
@dubielzyk-expensify Actualy I just realised that I tested the wrong branch (Was working on another bug) and attached videos for that in the PR. Will change that to the correct videos.

> The overflow menu isn't correct on desktop. It shouldn't replicate the overflow menu like mobile, but rather just show the missing options. See mock underneath:

Yes, I was aware of this and has been [discussed in slack here](https://expensify.slack.com/archives/C01GTK53T8Q/p1704887111830419) to be like this

**Comment by quinthar** [comment]
Who is this waiting on?  Can we merge today?

**Comment by dubielzyk-expensify** [comment]
Everything looks good except for the overflow on desktop. We're still seeing the full overflow menu:
CleanShot 2024-01-15 at 09 06 30@2x

It should look like this:
image

**Comment by esh-g** [comment]
@dubielzyk-expensify Yes, I was aware of this and has been [discussed in slack here](https://expensify.slack.com/archives/C01GTK53T8Q/p1704887111830419) to be like this. Please bring your opinion to slack so that we can decide

**Comment by dubielzyk-expensify** [comment]
Thanks for that. Sorry, I was going through my GitHub issue before Slack this morning :) I've replied on the thread. Awaiting designers feedback, then we can charge ahead with the final decision. Thanks for raising this, @esh-g 👍

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by rushatgabhane — APPROVED** [review]
LGTM!

@esh-g @dubielzyk-expensify i think we can do [that](https://github.com/Expensify/App/pull/34031#issuecomment-1891103980) in a follow up PR after the discussion is done, it should be a small change. 

So we can merge this PR. What do you think?

**Review by dubielzyk-expensify — DISMISSED** [review]

**Comment by mountiny** [comment]
@esh-g can you resolve merge conflicts and retest afterwards please?

@rushatgabhane  could you give it a quick test after that too please?

**Comment by quinthar** [comment]
Looks like this is going great -- can we get this merged today so we can deploy it to staging tomorrow?

**Comment by esh-g** [comment]
Fixing conflicts now

**Comment by esh-g** [comment]
@rushatgabhane @mountiny I encountered a bug that when you select the three dot menu with keyboard, it shows up in the corner. I tried to prevent this but I think this is because the `pageX` and `pageY` are set to `0` in `KeyboardEvent` which is what the `PopoverContextMenu` relies on for it's position. Should we look for a solution to this or is this not a priority?

https://github.com/Expensify/App/assets/77237602/a8e1ccc5-3fae-4490-a114-96bafa3cdc51

**Comment by mountiny** [comment]
I dont think we have to hold for this we are still not focusing on accessibility features

That being said i think you should aim for fix that in the following up pr in scope of this task as this is essentially an UX regression introduced by this change @esh-g

**Review by mountiny — APPROVED** [review]
Thanks!

**Pull request merged**
